Clarify statement blocks macro  need ref (#5138)
fishtown-analytics/docs.getdbt.com: website/docs/reference/dbt-jinja-functions/statement-blocks.md

statement(name=None, fetch_result=False, auto_begin=True)
```

> When executing a `statement`, dbt needs to understand how to resolve references to other dbt models or resources. You an explicitly specify it using `-- depends_on` or doing a `ref()` function outside the statement block ensure proper execution.

Author: ```suggestion
When executing a `statement`, dbt needs to understand how to resolve references to other dbt models or resources. You can explicitly specify it using `-- depends_on` or doing a `ref()` function outside the statement block to ensure proper execution.
```

> When executing a `statement`, dbt needs to understand how to resolve references to other dbt models or resources. You can explicitly specify it using `-- depends_on` or doing a `ref()` function outside the statement block to ensure proper execution.

Reviewer: ```suggestion
When executing a `statement`, dbt needs to understand how to resolve references to other dbt models or resources. If you are already `ref`ing the model outside of the statement block, the dependency will be automatically inferred, but otherwise you will need to [force the dependency](/reference/dbt-jinja-functions/ref#forcing-dependencies) with `-- depends_on`.
```

<expandable alt_header="Example using -- depends_on">

```sql
-- depends_on: {{ ref('users') }}

> {%- call statement('states', fetch_result=True) -%}
> 
>     select distinct state from {{ ref('users') }}
> 
> {%- endcall -%}

Reviewer: ```suggestion
{% call statement('states', fetch_result=True) -%}

    select distinct state from {{ ref('users') }}

    /*
    The unique states are: {{ load_result('states')['data'] }}
    */
{%- endcall %}
```

Suggest a bit less whitespace control to respect any other newlines in the user's code, and demonstrate how to interact with the result of the query

```
</expandable>

<expandable alt_header="Example using ref() function">

```sql

> {%- call statement('states', fetch_result=True) -%}
> 
>     select distinct state from {{ ref('users') }}
> 
> {%- endcall -%}

Reviewer: ```suggestion
{% call statement('states', fetch_result=True) -%}

    select distinct state from {{ ref('users') }}

    /*
    The unique states are: {{ load_result('states')['data'] }}
    */

{%- endcall %}
```

Same here
